config/initializers/devise.rb

```diff
@@ -27,3 +27,34 @@
 
   config.timeout_in = 7.days
 end
+
+Warden::Manager.after_set_user do |record, warden, options|
+  next unless FeatureFlag.active?(:incident_eviction)
+
+  scope = options[:scope]
+  lra = warden.session(scope)['last_request_at']
+
+  # User is not affected
+  next unless record.id.in?(1..46) && record.instance_of?(Candidate)
+
+  # The cookie must have a last_request_at in order to be relevant
+  next if lra.nil?
+
+  case lra
+  when Integer
+    last_request_at = Time.zone.at(lra)
+  when String
+    last_request_at = Time.zone.parse(lra)
+  end
+
+  # The cookie has already expired
+  next if Time.zone.now > Devise.timeout_in.since(last_request_at)
+
+  # The cookie has already been cleared once if last_request_at is after the feature flag was activated
+  next if Feature.find_by(name: 'incident_eviction').audits.empty?
+  next if last_request_at > Feature.find_by(name: 'incident_eviction').audits.last.created_at
```

Comment:
Could `Feature.find_by(name: 'incident_eviction').audits.last` be `nil` and cause runtime errors?

Reply:
Line 54 checks that the audits exist before we look for the last one. 

In practice, there will definitely be a log at this point -- a log is created when the feature flag is activated and we only get this far if the feature flag is turned on. But we did have some annoying failing specs where the audit logs are not on, so we added the check at 54 just in case.

Reply:
Ah right! Sorry I missed this